dashboard/dashboard-service.js

```diff
@@ -0,0 +1,127 @@
+                        'questionnaire-document-version': '4.1.0',
+                        created: '2022-01-18 04:05:06+01',
+                        modified: '2022-01-18 05:05:06+01',
+                        state: 'NOT_STARTED',
+                        'user-id': userId
+                    }
+                },
+                {
+                    type: 'metadata',
+                    id: 'fac0786c-bb58-4dfc-b456-9d9a8aa46510',
+                    attributes: {
+                        'questionnaire-id': 'fac0786c-bb58-4dfc-b456-9d9a8aa46510',
+                        'questionnaire-document-version': '4.1.0',
+                        created: '2021-01-18 04:05:06+01',
+                        modified: '2021-01-18 05:05:06+01',
+                        state: 'NOT_STARTED',
+                        'user-id': userId
+                    }
+                }
+            ]
+        };
+    }
+
+    // stubbbing response from:
+    // GET /questionnaires/{questionnaireId}/sections/{sectionId}/answers << Get answer resource for a specific section
+    async function getSectionAnswersByQuestionnaireId(
+        questionnaireId // ,
+        // sectionId = 'p-applicant-enter-your-name' TODO: utilise this formal paramter in the function call in routes.js
+    ) {
+        if (questionnaireId === '285cb104-0c15-4a9c-9840-cb1007f098fb') {
+            return {
+                data: [
+                    {
+                        type: 'answers',
+                        id: 'p-applicant-enter-your-name',
+                        attributes: {
+                            'q-applicant-firstname': 'Tony',
+                            'q-applicant-lastname': 'Stark'
+                        }
+                    }
+                ]
+            };
+        }
+
+        return {
+            data: [
+                {
+                    type: 'answers',
+                    id: 'p-applicant-enter-your-name',
+                    attributes: {
+                        'q-applicant-firstname': 'Bruce',
+                        'q-applicant-lastname': 'Banner'
+                    }
+                }
+            ]
+        };
+    }
+
+    async function getFlatQuestionnaireMetadataByUserId(userId) {
+        const metadataCollection = await getQuestionnaireMetadataCollectionByUserId(userId);
+        return metadataCollection.data.map(metadatum => {
+            return {
+                questionnaireId: metadatum.attributes['questionnaire-id'],
+                modified: metadatum.attributes.modified
+            };
+        });
+    }
+
+    async function getTemplateData(userId) {
+        const dataset = [];
+        // eslint-disable-next-line no-restricted-syntax
+        for await (const flatMetadata of await getFlatQuestionnaireMetadataByUserId(userId)) {
+            const answers = await getSectionAnswersByQuestionnaireId(flatMetadata.questionnaireId);
```

Comment:
as discussed, the function signature has the `sectionId` using a default value to avoid having to pass it in. We are only needing the user's name, so the sectionId needed will never change.